Architecture.md

```diff
@@ -55,10 +61,62 @@
 
 ### Database architecture
 
-***TODO: [CLIENT] describe the database architecture, as referenced in the MCD in the README***
+As the database is unified and serves as a focal point of the application, everything is contained in the same logical database.
+The architecture is as follows:
+
+```mermaid
+erDiagram
+    USERS {
+        uuid id PK
+        varchar(40) username UK
+        varchar(100) email UK
+        varchar(255) password
+        bool is_admin "Default false"
+        uuid email_nonce UK "Nullable, default uuid()"
+        uuid password_nonce UK "Nullable"
+        timestamp created_at
+        timestamp updated_at "Nullable"
+    }
+    SSH_KEYS {
+        uuid id PK
+        text value "public SSH key content"
+        varchar(30) name
+        timestamp created_at
+        timestamp updated_at "Nullable"
+        uuid user_id FK
+    }
+    PROJECTS {
+        uuid id PK
+        varchar(40) name
+        jsonb config
+        timestamp created_at
+        timestamp updated_at "Nullable"
+        uuid user_id FK
+    }
+
+    USERS ||--o{ PROJECTS : manage
+    USERS ||--o{ SSH_KEYS : possess
+```
+
+As you can notice, every table has a uuid field as a primary key. Compared to `SERIAL` that is often used to identify a table, a uuid guarantees a better uniqueness across the database. A [Serial](https://www.postgresql.org/docs/current/datatype-numeric.html) may take up less space (4 bytes) than an [uuid](https://www.postgresql.org/docs/current/datatype-uuid.html) (16 bytes) but since it is a series of incrementing integers it offers information about the time of creation and makes it easier to guess the id whereas an uuid is generated randomly and nearly impossible to duplicate.
+
+Furthermore, since the projects will be deployed at a subdomain that is named after `https://{projectId}.user-app.paastech.cloud` we needed to hide the internal database structure that emerges with a serial id. Therefore, using a uuid prevents sharing sensitive data with everyone and prevents targeted attacks.
+
+
+The table `users` contains all the necessary information about each user. Upon user creation, an `email_nonce` is automatically created and a user account is only considered active once the email has been confirmed and the field `email_nonce` is null.
+Should the user wish to reset his password, the field `password_nonce` will contain a unique uuid allowing the user to reset his password.
+One user can have multiple projects or SSH keys.
+
+The `ssh_keys` specified by each user allow them to push their repository onto our git server. Each SSH key can have a name to make it easier to distinguish multiple keys, however, it is not required.
+A SSH key belongs to a user, so it will have the same permissions as the user on all his repositories at the moment.
```

Comment:
The phrase doesn't seem very clear. I hope I interpreted it correctly. And since SSH is pronounced as "Es-es-ash", we need to use "an" as an article.
```suggestion
An SSH key belongs to a user and not to a particular project at the moment. Thus, it will allow its owner to access any of their repositories. If a user has more than one SSH key, it does not matter which one they use to access their projects.
```